lib/TopSitesFeed.jsm

```diff
@@ -612,6 +612,48 @@ this.TopSitesFeed = class TopSitesFeed {
     this._broadcastPinnedSitesUpdated();
   }
 
+  updatePinnedSearchShortcuts({selectedShortcuts}) {
+    const {rows} = this.store.getState().TopSites;
+
+    // Unpin any shortcuts that are no longer selected.
+    rows.filter(row => row && row.isPinned && row.searchTopSite).forEach(({label, url, hostname}) => {
+      if (!selectedShortcuts.find(shortcut => shortcut.label === label)) {
+        NewTabUtils.pinnedLinks.unpin({url});
+        this.store.dispatch(ac.UserEvent({
```

Comment:
Obviously we'd want to keep this in content

Reply:
oh it needs to be from content? hmm then we'll have to have a bunch of duplicate logic about what we're pinning and unpinning... ? gah